### PR TITLE
Make libraries

### PR DESCRIPTION
**Pull request by darrylhodgins**
This is an attempt to make this project more in-line with how Particle Workbench handles external libraries.

So far, I have only moved `homekit` into a new library (`homekit-server`), and I've moved the other third-party libraries into `lib` as well.

I don't plan on moving _all_ of the accessory types into their own libraries, but ultimately that would make a clean separation of responsibilities.  One library for the base server, and one for each accessory type.  A specific project would then import only the necessary libraries.

I do have an example of a usage of `HomekitRelaySwitchAccessory` here: https://github.com/darrylhodgins/particle-homekit-relay-switch.

What do you think, Lukas?

Darryl

### CODE REVIEW

File: src/particle-hap.ino [hunk 1]
```diff
@@ -59,6 +65,7 @@ void setup() {
   hkServer->start();
 
   Particle.function("restart", restart);
+  Particle.function("resetAll", resetAll);
```
Comment (darrylhodgins) [inline src/particle-hap.ino:68]: I forgot I put this in here, but I think it's useful!  :)

**Comment by ljezny** [comment]
Hi, i follow your idea and basically like it. 
But I would prefer to have all in one repo (homekit-server and example usage in one repo). Third-party libs doesn't have to be in this repo or at least it should be in i.e example/temperature_sensor/lib folder. 

I am also thinking that *AccessoryDescriptor and *ServiceDescriptor classes should be included in homekit-server - bcs it seem they are very handy to describe services and accessories in object model.

 I am also thinking to hardly separate all samples.. 

I think i will refactor this weekend.

**Comment by darrylhodgins** [comment]
Oh, certainly.  All in one repo is fine.  I was just showing an example of how someone else might want to use the base `homekit-server`, and extend it for their own project.  :)

**Comment by ljezny** [comment]
Yes, I will refactor it, i understand the needs. But it seems you are the first, who actually wants to use it :) I am glad :) 
Please contact me directly ljezny@gmail.com <mailto:ljezny@gmail.com> so we can discuss it.

L.

**Comment by ljezny** [comment]
I've closed it. I've get the idea and did it myself.

**Pull request closed without merging**